oeps/processes/oep-0056-architectural-advisory-process.rst

```diff
@@ -0,0 +1,197 @@
+
+Advisory Process Artifacts
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The final artifact of the Advisory Process is an Architectural Decision Record document checked in to a repository relevant to the decision.  Prior to that, however, we suggest as a best practice using Github's Issues and Pull Request tooling to conduct the Advisory Process.  The advantage of doing so is that Issue/Pull Request comments remove the burden of recording who gave particular advice, and allow us to have threaded, focused conversations about various aspects of ADR.
+
+Note that this section is GitHub-specific because that's where the Open edX platform's code resides.  The idea of "issues" and "pull requests" is otherwise portable to other version control platforms.
+
+GitHub Issues (Optional)
+************************
+
+We would encourage decision makers to use a GitHub Issue to have initial conversations around the context of a decision, or to write down upcoming/potential decisions. It may be that - at the start of a project - we're not even sure what decisions need to be made.  They may not even reveal themselves until we're in the middle of coding.
+
+An Issue is a great place to have these initial, loose, conversations about a decision space without the burden of creating a full ADR (creating a new document in the right place, authoring the sections of an ADR, creating a branch, committing, creating a PR, etc.)  It may not be necessary for more well-defined or focused decisions.  We want to preserve this as an optional first step, as we recognize that creating a "formal" ADR can feel inexplicably daunting or like a distraction, especially when possible decisions come up in the course of working on an implementation.  The mental burden of breaking "flow state" and creating a new ADR document is often problematic.  We hope jotting some notes in an Issue as a starting point will allow engineers to record their potential decision point and quickly continue their work.
+
+GitHub Pull Requests
+********************
+
+Whether or not a GitHub Issue is used, a GitHub Pull Request will exist for creation/merging of the ADR.  This is the primary medium for the Advisory Process, and where we expect conversations will take place.  By the time a pull request is created, we expect the decision maker will have written down the context of the decision - necessary to orient advisors to the problem space - as well as draft of their potential decision.
+
+Decision Maker's Checklist (DMC)
+--------------------------------
+
+In order to help ADR authors ensure they get advice from the proper set of stakeholders, we will create a checklist of possible groups/individuals to choose from.  It's the responsibility of the author to ensure the relevant people are aware of the ADR and are given the opportunity to provide feedback.
```

Comment:
I'm trying to picture this checklist.  Roughly how many entries are you imagining on this list? 10? 50? 150?  I picture it like a Python dictionary, where the values are people or groups, and the keys are.. what?

Reply:
Admittedly my gut says we'll iterate on this a bit as we start trying to write it down, but as an initial proposal, I'd expect the keys to be domains of various kinds: Discovery, Ecommerce, LMS, Frontend Arch, XBlocks, UX, Discussions, Design System, Frontend Testing, IT, BTR, DEPR, etc.  I'm just making those up.  I think it'll require some curation and information architecture to make it a useful reference.

Reply:
I think it's also worth thinking about the different use cases for it - is it just a decision maker that's going to go look up stakeholders?  Will this also get used to find folks to ask technical questions?  Will it be mistaken for a maintainer spreadsheet?  IS it a maintainer spreadsheet?  I'm not sure we can fully define the details before we start iterating on it, but we may be able to at least say what we want from it.

Reply:
If there would be other usage of it (which is something I agree with), shouldn't we then call that document something else, e.g. subject matter experts SME... And then in this OEP, we could say one of the step of creating an ADR is to create or write a checklist of stakholder after going over the SME list?